app/src/lib/stores/helpers/branch-pruner.ts

```diff
@@ -159,9 +170,13 @@ export class BranchPruner {
 
     const gitStore = this.gitStoreCache.get(this.repository)
     for (const branch of branchesReadyForPruning) {
-      await gitStore.performFailableOperation(() =>
-        deleteBranch(this.repository, branch!, null, false)
+      const isDeleted = await gitStore.performFailableOperation(() =>
+        deleteBranch(this.repository, branch, null, false)
       )
+
+      if (isDeleted) {
+        log.info(`Deleted ${branch.name} - ${branch.tip.sha}`)
```

Comment:
@niik is this sufficient enough info to recover in the event of something going wrong?

Reply:
Yup, this will ensure it's possible to recreate a deleted branch